src/Infrastructure.Styles.Analyzer/RMSYN1103ExtensionMethodsTestedAsNormalInvocationAnalyzer.cs

```diff
@@ -0,0 +1,112 @@
+﻿// This file is part of the re-motion Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// re-motion Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Infrastructure.Styles.Analyzer
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class RMSYN1103ExtensionMethodsTestedAsNormalInvocationSyntaxAnalyzer : DiagnosticAnalyzer
+    {
+        public const string DiagnosticId = "RMSYN1103";
+
+        internal const string Title = "Extension methods are tested using normal method invocation syntax";
+        internal const string Message = "Extension methods should be tested with normal method invocation syntax";
+
+        internal const string Description =
+            "Extension methods should be tested using normal method invocation instead of by extension";
+
+        internal const string Category = "Readability";
+
+        private static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            DiagnosticId,
+            Title,
+            Message,
+            Category,
+            DiagnosticSeverity.Warning,
+            true,
+            Description);
+
+        private static readonly List<string> TestAttributeList = new List<string> { "Test", "TestAttribute" };
+
+        public override void Initialize (AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.EnableConcurrentExecution();
+
+            context.RegisterSyntaxNodeAction(AnalyzeInvocationExpression, SyntaxKind.InvocationExpression);
+        }
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } =
+            ImmutableArray.Create(Rule);
+
+        private static void AnalyzeInvocationExpression (SyntaxNodeAnalysisContext context)
+        {
+            var invocationExpression = context.Node as InvocationExpressionSyntax;
+
+            if (!IsInTestMethod(invocationExpression!)) return;
```

Comment:
Single statements in ifs the next line please. Please fix in the other files as well